R/last_connection.R

```diff
@@ -10,11 +10,20 @@
 #' @seealso \code{\link{postgresql_connection}}
 #' @export
 last_connection <- function() {
-  get_cache('last_connection') %||%
+  conn <- get_cache('last_connection')
+  if (is(conn, "error")) {
+    conn <- NULL
+  }
+  if (is(conn, 'PqConnection')) {
+    isValidConnection <- tryCatch(dbIsValid(dbSendQuery(conn, 'select 1')), error = function(e) FALSE)
```

Comment:
@kirillseva I actually tried ```dbGetQuery```, but the return is a ```data.frame``` whereas the return from ```dbSendQuery``` is a ```PqResult``` object (which is what ```dbIsValid``` wants). Check it out:

```R
> dbGetQuery(conn, 'select 1') -> a
> a
  ?column?
1        1
> class(a)
[1] "data.frame"
```

```R
> dbSendQuery(conn, 'select 1') -> b
> b
<PqResult>
  SQL  select 1
  ROWS Fetched: 0 [complete]
       Changed: 0
> class(b)
[1] "PqResult"
attr(,"package")
[1] "RPostgres"
```